### PR TITLE
[E2E] Recurring

### PR DESCRIPTION
**Pull request by tiagofilipenunes**

### CODE REVIEW

File: e2e/tests/strategy/recurring/edit.ts [hunk 1]
```diff
@@ -24,6 +29,30 @@ export const editPriceStrategyTest = (testCase: CreateStrategyTestCase) => {
...
+    const myStrategies = new MyStrategyDriver(page);
+    const strategyEdited = await myStrategies.getStrategy(1);
+
+    // Check range
+    const [buySetting, sellSetting] = getRecurringSettings(testCase);
+
+    const buyTooltip = await strategyEdited.priceTooltip('buy');
+    if (buySetting === 'limit') {
+      await expect(buyTooltip.price()).toHaveText(buy.min);
+      await expect(buyTooltip.price()).toHaveText(buy.max);
```
Comment (GrandSchtroumpf) [inline e2e/tests/strategy/recurring/edit.ts:41]: You don't need to verify twice. Here you are checking if buy.min === buy.max which is outside the scope of E2E
```suggestion
      await expect(buyTooltip.price()).toHaveText(buy.min);
```

File: e2e/tests/strategy/recurring/edit.ts [hunk 2]
```diff
@@ -24,6 +29,30 @@ export const editPriceStrategyTest = (testCase: CreateStrategyTestCase) => {
...
+    } else {
+      await expect(buyTooltip.minPrice()).toHaveText(buy.min);
+      await expect(buyTooltip.maxPrice()).toHaveText(buy.max);
+    }
+    await buyTooltip.waitForDetached();
+
+    const sellTooltip = await strategyEdited.priceTooltip('sell');
+    if (sellSetting === 'limit') {
+      await expect(sellTooltip.price()).toHaveText(sell.min);
+      await expect(sellTooltip.price()).toHaveText(sell.max);
```
Comment (GrandSchtroumpf) [inline e2e/tests/strategy/recurring/edit.ts:51]:
```suggestion
      await expect(sellTooltip.price()).toHaveText(sell.min);
```

File: e2e/tests/strategy/recurring/renew.ts [hunk 3]
```diff
@@ -24,6 +28,31 @@ export const renewStrategyTest = (testCase: CreateStrategyTestCase) => {
...
+    const strategyEdited = await myStrategies.getStrategy(1);
+
+    // Check range
+    const { buy, sell } = testCase.output.editPrice;
+    const [buySetting, sellSetting] = getRecurringSettings(testCase);
+
+    const buyTooltip = await strategyEdited.priceTooltip('buy');
+    if (buySetting === 'limit') {
+      await expect(buyTooltip.price()).toHaveText(buy.min);
+      await expect(buyTooltip.price()).toHaveText(buy.max);
```
Comment (GrandSchtroumpf) [inline e2e/tests/strategy/recurring/renew.ts:41]:
```suggestion
      await expect(buyTooltip.price()).toHaveText(buy.min);
```

File: e2e/tests/strategy/recurring/undercut.ts [hunk 4]
```diff
@@ -46,12 +47,28 @@ export const undercutStrategyTest = (testCase: CreateStrategyTestCase) => {
...
     await expect(strategyUndercut.budgetFiat('sell')).toHaveText(sell.fiat);
 
+    // Check range
+    const [buySetting, sellSetting] = getRecurringSettings(testCase);
+
     const buyTooltip = await strategyUndercut.priceTooltip('buy');
-    await expect(buyTooltip.startPrice()).toHaveText(buy.min);
+    if (buySetting === 'limit') {
+      await expect(buyTooltip.price()).toHaveText(buy.min);
+      await expect(buyTooltip.price()).toHaveText(buy.max);
```
Comment (GrandSchtroumpf) [inline e2e/tests/strategy/recurring/undercut.ts:56]:
```suggestion
      await expect(buyTooltip.price()).toHaveText(buy.min);
```

File: e2e/tests/strategy/recurring/undercut.ts [hunk 5]
```diff
@@ -46,12 +47,28 @@ export const undercutStrategyTest = (testCase: CreateStrategyTestCase) => {
...
+      await expect(buyTooltip.minPrice()).toHaveText(buy.min);
+      await expect(buyTooltip.maxPrice()).toHaveText(buy.max);
+    }
     await buyTooltip.waitForDetached();
 
     const sellTooltip = await strategyUndercut.priceTooltip('sell');
-    await expect(sellTooltip.startPrice()).toHaveText(sell.min);
+    if (sellSetting === 'limit') {
+      await expect(sellTooltip.price()).toHaveText(sell.min);
+      await expect(sellTooltip.price()).toHaveText(sell.max);
```
Comment (GrandSchtroumpf) [inline e2e/tests/strategy/recurring/undercut.ts:66]:
```suggestion
      await expect(sellTooltip.price()).toHaveText(sell.min);
```

File: e2e/tests/strategy/recurring/create.ts [hunk 6]
```diff
@@ -51,12 +53,41 @@ export const createRecurringStrategy = (testCase: CreateStrategyTestCase) => {
...
     await expect(strategy.budget('sell')).toHaveText(output.sell.budget);
     await expect(strategy.budgetFiat('sell')).toHaveText(output.sell.fiat);
+
+    // Check range
+    const [buySetting, sellSetting] = getRecurringSettings(testCase);
+
+    const buyTooltip = await strategy.priceTooltip('buy');
+    if (buySetting === 'limit') {
+      await expect(buyTooltip.price()).toHaveText(output.buy.min);
+      await expect(buyTooltip.price()).toHaveText(output.buy.max);
```
Comment (GrandSchtroumpf) [inline e2e/tests/strategy/recurring/create.ts:72]:
```suggestion
      await expect(buyTooltip.price()).toHaveText(output.buy.min);
```

File: e2e/tests/strategy/recurring/create.ts [hunk 7]
```diff
@@ -51,12 +53,41 @@ export const createRecurringStrategy = (testCase: CreateStrategyTestCase) => {
...
+    } else {
+      await expect(buyTooltip.minPrice()).toHaveText(output.buy.min);
+      await expect(buyTooltip.maxPrice()).toHaveText(output.buy.max);
+    }
+    await buyTooltip.waitForDetached();
+
+    const sellTooltip = await strategy.priceTooltip('sell');
+    if (sellSetting === 'limit') {
+      await expect(sellTooltip.price()).toHaveText(output.sell.min);
+      await expect(sellTooltip.price()).toHaveText(output.sell.max);
```
Comment (GrandSchtroumpf) [inline e2e/tests/strategy/recurring/create.ts:82]:
```suggestion
      await expect(sellTooltip.price()).toHaveText(output.sell.min);
```

File: e2e/tests/strategy/recurring/create.ts [hunk 8]
```diff
@@ -51,12 +53,41 @@ export const createRecurringStrategy = (testCase: CreateStrategyTestCase) => {
...
+    } else {
+      await expect(sellTooltip.minPrice()).toHaveText(output.sell.min);
+      await expect(sellTooltip.maxPrice()).toHaveText(output.sell.max);
+    }
+    await sellTooltip.waitForDetached();
+
+    await notif.close;
```
Comment (GrandSchtroumpf) [inline e2e/tests/strategy/recurring/create.ts:89]: This is a method
```suggestion
    await notif.close();
```

File: e2e/tests/strategy/recurring/edit.ts (hunk 1 above)
Comment (tiagofilipenunes) [reply e2e/tests/strategy/recurring/edit.ts:41]: agreed

**Review by GrandSchtroumpf — APPROVED** [review]
Very nice 
ApplaudingYesGIF

**Pull request merged**